### PR TITLE
Added a try/catch to StorageEmulator.TryStart()

### PR DESCRIPTION
**Pull request by sergeybykov**
Process.Start was throwing in our nightly tests breaking them. This is a replacement for https://github.com/dotnet/orleans/pull/464.

### CODE REVIEW

File: src/OrleansTestingHost/StorageEmulator.cs [hunk 1]
```diff
@@ -119,18 +116,21 @@ public static bool TryStart()
...
                 using(var process = Process.Start(CreateProcessArguments("start")))
                 {
                     process.WaitForExit();
-                    isSuccess = process.ExitCode == 0;
+                    return process.ExitCode == 0;
                 }
             }
-
-            return isSuccess;
+            catch (Exception)
```
Comment (veikkoeeva) [inline src/OrleansTestingHost/StorageEmulator.cs]: This could be plain ``catch``.

File: src/OrleansTestingHost/StorageEmulator.cs [hunk 2]
```diff
@@ -140,18 +140,21 @@ public static bool Start()
...
                 using(var process = Process.Start(CreateProcessArguments("stop")))
                 {
                     process.WaitForExit();
-                    isSuccess = process.ExitCode == 0;
+                    return process.ExitCode == 0;
                 }
             }
-
-            return isSuccess;
+            catch (Exception)
```
Comment (veikkoeeva) [inline src/OrleansTestingHost/StorageEmulator.cs]: Plain ``catch``.

**Comment by veikkoeeva** [comment]
Your refactoring simplified and made it more robust. Could you share what was the exception? Just out of curiosity?

(Non-related, I got overwhelmingly busy some hours ago and this might take the whole weekend. A major performance regression in production. So I might be unavailable etc. :))

File: src/OrleansTestingHost/StorageEmulator.cs (hunk 1 above)
Comment (sergeybykov) [inline src/OrleansTestingHost/StorageEmulator.cs]: Good point. I was too lazy to clean up after ReSharper. :-) Updated.

**Comment by sergeybykov** [comment]
@veikkoeeva It was "System.ComponentModel.Win32Exception: The system cannot find the file specified".

Good luck with finding root cause of the regression. I hope it won't ruin your whole weekend.

**Comment by veikkoeeva** [comment]
@sergeybykov And I thought I checked that. This has been an all around clammy week. Thanks for picking up this, looks good to me (heh, maybe I shouldn't be too sure :)).

(The root cause is SAN, to some extent, and there's some strange CPU wobbling, maybe due to virtual machine infrastructure. Other people investigate now.)

**Pull request merged**